vm/src/cformat.rs

```diff
@@ -113,7 +114,7 @@ impl CConversionFlags {
 
 #[derive(Debug, PartialEq)]
 enum CFormatQuantity {
-    Amount(usize),
+    Amount(isize),
```

Comment:
In the code `"%*s" % (-5, "spam")`, the Amount comes through as `-5` for the minimum width, and in `"%.*d" % (-5, "spam")` the precision does (albeit meaninglessly IIRC). In the width case, a negative implies left-adjusting to that width rather than right-adjusting.

We could potentially determine this earlier and store a tuple of absolute width and adjustment instead?

Reply:
OK, I've stripped this fix back considerably: it calls `.abs()` in the two places `::Amount` is instantiated, and uses `saturating_sub` to avoid another overflow condition. This still isn't quite correct for some corner cases, but it's certainly better than an OOM crash in non-corner cases.

I have some code locally to handle using negative Amounts to perform the appropriate padding, based on my `isize` implementation: that should address those corner cases. I'll look at reworking it to avoid stringing isize's throughout the code.